### DISCUSSION TITLE
openidc config for frontend https + backend http

### DISCUSSION THREAD

**Discussion by it-guru** [OpenIDC/mod_auth_openidc#1212] [Ideas]
If you have your application server (Apache+openidc) behind a reverse proxy, which terminates the https session and the communication to the application server is done by http, you get a problem.
You get an error like this ...
__oidc_authenticate_user: the URL scheme (https) of the configured OIDCRedirectURI does not match 
the URL scheme of the URL being accessed (http): the "state" and "session" cookies will not be shared between the two!__
... while configuring openidc.

The problem is, openidc needs to consider the HTTP_FRONT_END_HTTPS variable around the code sequence ...
https://github.com/OpenIDC/mod_auth_openidc/blob/master/src/handle/request.c#L60

Maybe a simple switch to disable the scheme check, will also help to use openid in such environments.

**Comment by zandbelt** [comment]
see https://github.com/OpenIDC/mod_auth_openidc/wiki#8-how-do-i-run-mod_auth_openidc-behind-a-reverse-proxy
